crates/mapdiffbot2/src/rendering.rs

```diff
@@ -282,89 +317,260 @@ pub fn render_map(
         max: (bounds.right, bounds.top),
         render_passes,
         errors,
-        bump: &bump,
+        arena: &arena,
     };
     minimap::generate(minimap_context, icon_cache)
         .map_err(|_| eyre::anyhow!("An error occured during map rendering"))
 }
 
+#[derive(Clone, Copy)]
+pub enum MapType {
+    Head,
+    Base,
+}
+
+pub type RenderedMaps = IndexMap<PathBuf, Vec<u8>, ahash::RandomState>;
+
 pub fn render_map_regions(
     context: &RenderingContext,
-    maps: &[&MapWithRegions],
+    maps: &[(&str, &MapWithRegions)],
     render_passes: &[Box<dyn RenderPass>],
-    output_dir: &Path,
+    (output_dir, blob_client): (&Path, Azure),
     filename: &str,
     errors: &RenderingErrors,
-) -> Result<()> {
+    map_type: MapType,
+) -> Result<RenderedMaps> {
     let objtree = &context.obj_tree;
     let icon_cache = &context.icon_cache;
-    let _: Result<()> = maps
+    let results = maps
         .par_iter()
-        .enumerate()
-        .map(|(idx, map)| {
-            for z_level in 0..map.map.dim_z() {
-                if let Some(bounds) = map
-                    .bounding_boxes
-                    .get(z_level)
-                    .expect("No bounding box generated for z-level")
-                {
-                    let image = render_map(
+        .map(|(map_name, map)| -> Result<RenderedMaps> {
+            render_map_region(
+                map_name,
+                map,
+                map_type,
+                (objtree, icon_cache, errors, render_passes),
+                (output_dir, blob_client.clone(), filename),
+            )
+        })
+        .collect::<Vec<_>>();
+
+    results.iter().for_each(|res| {
+        if let Err(e) = res {
+            log::error!("{e:?}") //errors please
+        }
+    });
+
+    Ok(results
+        .into_iter()
+        .filter_map(|thing| thing.ok())
+        .flatten()
+        .collect::<RenderedMaps>())
+}
+
+fn render_map_region(
+    map_name: &str,
+    map: &MapWithRegions,
+    map_type: MapType,
+    (objtree, icon_cache, errors, render_passes): (
+        &ObjectTree,
+        &IconCache,
+        &RenderingErrors,
+        &[Box<dyn RenderPass>],
+    ),
+    (output_dir, blob_client, filename): (&Path, Azure, &str),
+) -> Result<RenderedMaps> {
+    let mut return_map: RenderedMaps = Default::default();
+    for z_level in 0..map.map.dim_z() {
```

Comment:
This doesn't render z-levels in parallel, is that desirable?

Reply:
Yes, I want it to fail and cancel out instead of continuing to render